modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/options/OptionsDDMFormFieldValueAccessorTest.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.form.field.type.internal.options;
+
+import com.liferay.dynamic.data.mapping.test.util.DDMFormValuesTestUtil;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.json.JSONFactoryImpl;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Rodrigo Paulino
+ */
+@PrepareForTest(LanguageUtil.class)
+@RunWith(PowerMockRunner.class)
+public class OptionsDDMFormFieldValueAccessorTest extends PowerMockito {
+
+	@BeforeClass
+	public static void setUpClass() {
+		_setUpJSONFactory(new JSONFactoryImpl());
+	}
+
+	@Before
+	public void setUp() {
+		_setUpLanguageUtil();
+	}
```

Comment:
> why did you decide to organize the methods this way? why the _setUpLanguageUtil was not included in setUpClass(), for example?

I tried setting up the `LanguageUtil` in the `setUpClass`, but, for some reason, the static class wasn't being mocked for every test. I didn't bother trying to look for the actual reason. Instead, I went the easier and quicker route that was to include a `setUp` method for it. Let me know if you think it's valid to investigate further.

> why did you pass the new JSONFactoryImpl() as a parameter instead of declaring it inside the _setUpJSONFactory

Just to buy one line. 😛

Reply:
Got it! No need to investigate further, in fact I would do exactly the same 😛